core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/kurtosis_plan_instruction/kurtosis_plan_instruction_capabilities.go

```diff
@@ -16,5 +17,7 @@ type KurtosisPlanInstructionCapabilities interface {
 
 	Execute(ctx context.Context, arguments *builtin_argument.ArgumentValuesSet) (string, error)
 
-	TryResolveWith(instructionsAreEqual bool, other KurtosisPlanInstructionCapabilities, enclaveComponents *enclave_structure.EnclaveComponents) enclave_structure.InstructionResolutionStatus
+	TryResolveWith(instructionsAreEqual bool, other *enclave_plan_persistence.EnclavePlanInstruction, enclaveComponents *enclave_structure.EnclaveComponents) enclave_structure.InstructionResolutionStatus
+
+	GetPersistableAttributes() (string, []string, []string, []string)
```

Comment:
WDYT about adding a comment mentioning what kind of info should the method be returning.
```suggestion
         // GetPersistableAttributes returns these instructions attributes (in case of existence): "instructionType", "starlarkCode", "serviceNames", "filesArtifactNames", "filesArtifactMd5s"
	GetPersistableAttributes() (string, []string, []string, []string)
```

Reply:
I agree, TBH I don't like this endless list of returned value. I'll see if I can wrap all this into an object with explicit attribute names